store/postgres/src/relational.rs

```diff
@@ -650,7 +650,9 @@ impl Layout {
         let table = self.table_for_entity(&entity_type)?;
         let entity_keys: Vec<&EntityKey> = entities.iter().map(|(key, _)| key).collect();
         ClampRangeQuery::new(table, &entity_type, &entity_keys, block).execute(conn)?;
-        Ok(InsertQuery::new(table, entity_type, entities, block)?.execute(conn)?)
+        Ok(InsertQuery::new(table, entity_type, entities, block)?
+            .get_results(conn)
+            .map(|ids| ids.len())?)
```

Comment:
You're absolutely right.

At first, I thought that `execute` was returning 0 because of our custom queries, but after close inspection, I noticed that that zero was coming from a mistake I'd made when obtaining the length of a drained vector... 

I'll revert those changes and remove the `LoadQuery` trait implementation for `ClampRangeQuery`.